src/db/migrations/20-add-consumable-views.ts

```diff
@@ -62,13 +62,22 @@ export const up = async (knex: Knex) => {
     await knex.schema.renameTable(oldName, newName);
   }
 
-  // specify conditions to add to each view
-  const conditions: {
-    [table in Table]?: string
+  // specify override sql to for creating a view
+  const overrides: {
+    [table in Table]?: string;
   } = {
     [Table.processedTracks]: `
-    "lossyArtworkIPFSHash" is not null 
-    and "lossyAudioIPFSHash" is not null`,
+    select t.* 
+    from "${Table.processedTracks}" t
+    join "${Table.ipfsPins}" p
+    on t."lossyAudioIPFSHash" = p.id 
+    join "${Table.ipfsPins}" p1
+    on t."lossyArtworkIPFSHash" = p1.id 
+    where "lossyArtworkIPFSHash" is not null 
+    and "lossyAudioIPFSHash" is not null
+    and p.status = 'pinned'
+    and p1.status = 'pinned'
```

Comment:
status isn't reliable (we don't check and update it after pinning) so lets leave that out for now